src/main/java/github/grace000/erversay/ServerThread.java

```diff
@@ -2,31 +2,71 @@
 
 import java.io.*;
 import java.net.*;
+import java.util.HashMap;
 
 public class ServerThread extends Thread {
     private Socket socket;
+    private InputStream inputStream;
+    private RequestParser parser;
+    private RequestReader reader;
+    private ResponseBuilder responseBuilder;
+    private RequestRouter router;
 
     public ServerThread(Socket socket) {
         this.socket = socket;
+        parser = new RequestParser();
+        reader = new RequestReader();
+        responseBuilder = new ResponseBuilder();
+        router = new RequestRouter();
+
     }
 
     public void run() {
-        try {
-
-            OutputStream outputFromServer = socket.getOutputStream();
-            PrintWriter serverPrintOut = new PrintWriter(new OutputStreamWriter(outputFromServer, "UTF-8"), true);
-
-            String response = new Response().build();
-            serverPrintOut.println(response);
-            serverPrintOut.flush();
+        String input = convertInputToString();
+        String[] request = parser.parse(input);
 
-            System.out.println("Message sent");
+        HashMap routedRequest = router.route(request);
+        String response = responseBuilder.getResponse(routedRequest);
+        writeResponse(socket, response);
 
+        System.out.println("Message sent");
+        try {
+            socket.close();
         } catch (IOException e) {
             System.out.println("Server exception: " + e.getMessage());
             e.printStackTrace();
         }
     }
 
+    public void writeResponse(Socket socket, String response) {
+        try {
+            OutputStream outputFromServer = socket.getOutputStream();
+            PrintWriter output = new PrintWriter(new OutputStreamWriter(outputFromServer, "UTF-8"), true);
+            output.println(response);
+            output.flush();
+        } catch (IOException e) {
+            System.out.println("WRITE RESPONSE ERROR");
+            e.printStackTrace();
+        }
+    }
 
+    public String convertInputToString() {
+        socketGetInputStream();
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+        String convertedInput = "";
+        try {
+            convertedInput = reader.readRequest(bufferedReader);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Control flow in Java can be weird if you aren't used to exceptions. You need to learn how to use exceptions to indicate to the method caller that something went wrong. In this case, you failed to read data from the input stream. Instead of returning an empty string to the caller (who tries to parse it) you should allow this exception to be thrown, and handled by the calling code (which can close the connection and re-enter the `accept` loop).

Reply:
I'll make a card for this.